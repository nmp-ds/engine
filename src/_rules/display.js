import { handler as h } from '#utils';

export const display = [
  ['flex', { display: 'flex' }],
  ['inline-flex', { display: 'inline-flex' }],
  ['grid', { display: 'grid' }],
  ['inline-grid', { display: 'inline-grid' }],
  ['inline', { display: 'inline' }],
  ['block', { display: 'block' }],
  ['inline-block', { display: 'inline-block' }],
  ['contents', { display: 'contents' }],
  ['flow-root', { display: 'flow-root' }],
  ['list-item', { display: 'list-item' }],
  ['hidden', { display: 'none' }],
  ['inline-table', { display: 'inline-table' }],
  ['table', { display: 'table' }],
  ['table-caption', { display: 'table-caption' }],
  ['table-cell', { display: 'table-cell' }],
  ['table-column', { display: 'table-column' }],
  ['table-column-group', { display: 'table-column-group' }],
  ['table-footer-group', { display: 'table-footer-group' }],
  ['table-header-group', { display: 'table-header-group' }],
  ['table-row', { display: 'table-row' }],
  ['table-row-group', { display: 'table-row-group' }],
  // unset, revert, inherit
  [/^display-(.+)$/, ([, c]) => ({ display: h.bracket.cssvar.global(c) || c })],
];
