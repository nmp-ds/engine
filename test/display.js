import { describe, expect, test } from 'vitest';
import { setup } from './_helpers.js';
import { globalKeywords } from '#utils';
import * as displayRules from '../src/_rules/display.js';


setup();

describe('display', () => {
  test('should render styles for static classes', async ({ uno }) => {
    const staticClasses = Object.values(displayRules).flat().filter(rule => typeof rule[0] === 'string').map(rule => rule[0]);

    const { css } = await uno.generate(staticClasses);
    expect(css).toMatchSnapshot();
  });
  test('should render styles for arbitrary values', async ({ uno }) => {
    const arbitraryClasses = globalKeywords.map((key) => `display-${key}`);

    const { css } = await uno.generate(arbitraryClasses);
    expect(css).toMatchSnapshot();
  });
  test('should not render styles for invalid classes', async ({ uno }) => {
    const arbitraryClasses = ['unset','display-[revert]',  'display-[unset]',  'display-[inherit]'];

    const { css } = await uno.generate(arbitraryClasses);
    expect(css).toMatchInlineSnapshot('""');
  });
});

