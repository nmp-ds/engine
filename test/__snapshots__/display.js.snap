// Vitest Snapshot v1

exports[`display > should render styles for arbitrary values 1`] = `
"/* layer: default */
.display-inherit{display:inherit;}
.display-initial{display:initial;}
.display-revert{display:revert;}
.display-revert-layer{display:revert-layer;}
.display-unset{display:unset;}"
`;

exports[`display > should render styles for static classes 1`] = `
"/* layer: default */
.flex{display:flex;}
.inline-flex{display:inline-flex;}
.grid{display:grid;}
.inline-grid{display:inline-grid;}
.inline{display:inline;}
.block{display:block;}
.inline-block{display:inline-block;}
.contents{display:contents;}
.flow-root{display:flow-root;}
.list-item{display:list-item;}
.hidden{display:none;}
.inline-table{display:inline-table;}
.table{display:table;}
.table-caption{display:table-caption;}
.table-cell{display:table-cell;}
.table-column{display:table-column;}
.table-column-group{display:table-column-group;}
.table-footer-group{display:table-footer-group;}
.table-header-group{display:table-header-group;}
.table-row{display:table-row;}
.table-row-group{display:table-row-group;}"
`;
